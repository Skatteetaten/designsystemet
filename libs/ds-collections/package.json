{
  "name": "@skatteetaten/ds-collections",
  "groupId": "no.skatteetaten.aurora",
  "version": "1.4.0",
  "license": "Apache-2.0",
  "homepage": "https://www.skatteetaten.no/stilogtone/designsystemet/",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/Skatteetaten/designsystemet.git"
  },
  "publishConfig": {
    "registry": "https://nexus.sits.no/repository/npm-internal/"
  },
  "peerDependencies": {
    "i18next": "^23",
    "react": "^18",
    "react-i18next": "^14",
    "@skatteetaten/ds-core-utils": "1.3.0",
    "@skatteetaten/ds-icons": "1.3.0",
    "@skatteetaten/ds-buttons": "1.3.0",
    "@skatteetaten/ds-content": "1.3.0",
    "@skatteetaten/ds-typography": "1.3.0",
    "@skatteetaten/ds-progress": "1.3.0"
  },
  "scripts": {}
}
