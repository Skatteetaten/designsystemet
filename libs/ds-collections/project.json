{
  "name": "ds-collections",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/ds-collections/src",
  "projectType": "library",
  "tags": [],
  "implicitDependencies": ["!ds-dev-config"],
  "targets": {
    "lint": {
      "executor": "@nx/eslint:lint",
      "outputs": ["{options.outputFile}"]
    },
    "build": {
      "executor": "@nx/rollup:rollup",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/libs/ds-collections",
        "tsConfig": "libs/ds-collections/tsconfig.lib.json",
        "project": "libs/ds-collections/package.json",
        "entryFile": "libs/ds-collections/src/index.ts",
        "external": ["react/jsx-runtime"],
        "rollupConfig": ["libs/ds-collections/rollup.config.js"],
        "compiler": "swc",
        "assets": [
          {
            "glob": "libs/ds-collections/README.md",
            "input": ".",
            "output": "."
          },
          {
            "glob": "LICENSE",
            "input": ".",
            "output": "."
          }
        ],
        "babelUpwardRootMode": true
      }
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "libs/ds-collections/jest.config.ts"
      }
    },
    "stylelint": {
      "executor": "nx-stylelint:lint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": [
          "libs/ds-collections/**/*.css",
          "libs/ds-collections/**/*.scss"
        ]
      }
    },
    "tsc": {
      "executor": "@skatteetaten/ds-dev-config:tsCheck",
      "options": {
        "tsConfig": ["tsconfig.lib.json", "tsconfig.spec.json"]
      },
      "dependsOn": [
        {
          "projects": "ds-core-designtokens",
          "target": "build"
        }
      ]
    },
    "set-version": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npm version {args.packageVersion} --no-git-tag",
        "cwd": "libs/ds-collections"
      }
    },
    "publish": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npm publish dist/libs/ds-collections {args.switches}",
        "cwd": ""
      }
    }
  }
}
