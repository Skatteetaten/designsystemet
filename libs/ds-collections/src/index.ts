export * from './Accordion/Accordion';
export * from './Accordion/Accordion.types';
export * from './AccordionItem/AccordionItem.types';
export * from './Chips/Chips';
export * from './Chips/Chips.types';
export * from './ChipsToggle/ChipsToggle';
export * from './ChipsToggle/ChipsToggle.types';
export * from './ChipsRemovable/ChipsRemovable';
export * from './ChipsRemovable/ChipsRemovable.types';
export * from './OpenClose/OpenClose';
export * from './OpenClose/OpenClose.types';
export * from './StepList/StepList';
export * from './StepList/StepList.types';
export * from './StepListStep/StepListStep.types';
export * from './Tabs/Tabs';
export * from './Tabs/Tabs.types';
