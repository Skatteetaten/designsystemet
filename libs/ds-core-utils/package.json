{
  "name": "@skatteetaten/ds-core-utils",
  "groupId": "no.skatteetaten.aurora",
  "version": "1.1.1",
  "homepage": "https://www.skatteetaten.no/stilogtone/designsystemet/",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/Skatteetaten/designsystemet.git"
  },
  "publishConfig": {
    "registry": "https://nexus.sits.no/repository/npm-internal/"
  },
  "peerDependencies": {
    "i18next": "^23",
    "react": "^18",
    "react-i18next": "^14"
  },
  "scripts": {}
}
