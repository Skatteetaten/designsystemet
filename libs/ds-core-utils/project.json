{
  "name": "ds-core-utils",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/ds-core-utils/src",
  "projectType": "library",
  "tags": [],
  "implicitDependencies": ["!ds-dev-config"],
  "targets": {
    "build": {
      "executor": "@nx/rollup:rollup",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/libs/ds-core-utils",
        "tsConfig": "libs/ds-core-utils/tsconfig.lib.json",
        "project": "libs/ds-core-utils/package.json",
        "entryFile": "libs/ds-core-utils/src/index.ts",
        "external": ["react/jsx-runtime"],
        "rollupConfig": ["libs/ds-core-utils/rollup.config.cjs"],
        "compiler": "swc",
        "assets": [
          {
            "glob": "libs/ds-core-utils/README.md",
            "input": ".",
            "output": "."
          },
          {
            "glob": "LICENSE",
            "input": ".",
            "output": "."
          }
        ],
        "babelUpwardRootMode": true
      }
    },
    "lint": {
      "executor": "@nx/eslint:lint",
      "outputs": ["{options.outputFile}"]
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/libs/ds-core-utils"],
      "options": {
        "jestConfig": "libs/ds-core-utils/jest.config.ts"
      }
    },
    "stylelint": {
      "executor": "nx-stylelint:lint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": [
          "libs/ds-core-utils/**/*.css",
          "libs/ds-core-utils/**/*.scss"
        ]
      }
    },
    "tsc": {
      "executor": "@skatteetaten/ds-dev-config:tsCheck",
      "options": {
        "tsConfig": ["tsconfig.lib.json", "tsconfig.spec.json"]
      },
      "dependsOn": [
        {
          "projects": "ds-core-designtokens",
          "target": "build"
        }
      ]
    },
    "set-version": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npm version {args.packageVersion} --no-git-tag",
        "cwd": "libs/ds-core-utils"
      }
    },
    "publish": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npm publish dist/libs/ds-core-utils {args.switches}"
      }
    }
  }
}
