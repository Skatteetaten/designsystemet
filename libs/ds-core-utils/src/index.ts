export * from './SkatteetatenLogo/SkatteetatenLogo';
export * from './base-props.types';
export * from './defaults';
export * from './ExternalLayout/ExternalLayout';
export * from './InternalLayout/InternalLayout';
export * from './i18n';
export * from './Separator/Separator';
export * from './hooks/hooks';
export * from './formatter/formatter';
