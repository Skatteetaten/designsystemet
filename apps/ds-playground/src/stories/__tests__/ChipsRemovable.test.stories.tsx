import { Meta, StoryObj } from '@storybook/react';
import { expect, within } from '@storybook/test';

import { Chips, ChipsRemovableProps } from '@skatteetaten/ds-collections';
import { dsI18n } from '@skatteetaten/ds-core-utils';

const elementId = 'htmlId';

const meta = {
  component: Chips.Removable,
  title: 'Tester/Chips/Removable',
  argTypes: {
    // Baseprops
    key: { table: { disable: true } },
    ref: { table: { disable: true } },
    className: { table: { disable: true } },
    id: { table: { disable: true } },
    lang: { table: { disable: true } },
    'data-testid': { table: { disable: true } },
    // Props
    size: { table: { disable: true } },
    children: { table: { disable: true } },
    // HTML
    disabled: { table: { disable: true } },
    // Events
    onClose: { table: { disable: true } },
  },
} satisfies Meta<typeof Chips.Removable>;
export default meta;
type Story = StoryObj<typeof meta>;

const defaultArgs: ChipsRemovableProps = {
  children: 'Trondheim',
};

export const WithRef = {
  name: 'With Ref (FA1)',
  args: {
    ...defaultArgs,
    ref: (instance: HTMLButtonElement | null): void => {
      if (instance) {
        instance.id = 'dummyIdForwardedFromRef';
      }
    },
  },
  argTypes: {
    ref: { table: { disable: false } },
  },
  parameters: {
    imageSnapshot: { disable: true },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const container = canvas.getByRole('button');
    await expect(container).toHaveAttribute('id', 'dummyIdForwardedFromRef');
  },
} satisfies Story;

export const WithAttributes = {
  name: 'With Attributes(FA2-5)',
  args: {
    ...defaultArgs,
    id: elementId,
    className: 'dummyClassname',
    lang: 'nb',
    'data-testid': '123ID',
  },
  argTypes: {
    id: { table: { disable: false } },
    className: { table: { disable: false } },
    lang: { table: { disable: false } },
    'data-testid': { table: { disable: false } },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const container = canvas.getByRole('button');
    await expect(container).toHaveClass('dummyClassname');
    await expect(container).toHaveAttribute('id', elementId);
    await expect(container).toHaveAttribute('lang', 'nb');
    await expect(container).toHaveAttribute('data-testid', '123ID');
  },
} satisfies Story;

export const Defaults = {
  name: 'Defaults (A2, A3, B3 delvis)',
  args: {
    ...defaultArgs,
  },
} satisfies Story;

export const WithSizeSmall = {
  name: 'With Size Small (A1)',
  args: {
    size: 'small',
    ...defaultArgs,
  },
  argTypes: {
    size: { table: { disable: false } },
  },
} satisfies Story;

export const WithCloseIconAltText = {
  name: 'With Close Icon Alternative Text (B3 delvis)',
  args: {
    ...defaultArgs,
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const icon = canvas.getByTitle(
      dsI18n.t('ds_collections:chips.RemoveFilter')
    );
    await expect(icon).toBeInTheDocument();
  },
} satisfies Story;
