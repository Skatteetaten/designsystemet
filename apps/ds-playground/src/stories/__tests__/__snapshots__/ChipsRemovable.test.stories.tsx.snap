// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tester/Chips/Removable Defaults smoke-test 1`] = `
<div data-test-block="true">
  <button type="button"
          class="_chip_e7km3_5 _chip_removable_e7km3_62"
  >
    Trondheim
    <svg class="_systemIcon_6zbcy_2 _systemIcon_small_6zbcy_5 _icon_e7km3_77"
         viewbox="0 0 24 24"
         focusable="false"
         aria-labelledby="svgtitle-:r2:"
         aria-hidden="false"
         role="img"
    >
      <title id="svgtitle-:r2:">
        Fjern filter TODO
      </title>
      <path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z">
      </path>
    </svg>
  </button>
</div>
`;

exports[`Tester/Chips/Removable WithAttributes play-test 1`] = `
<div data-test-block="true">
  <button type="button"
          id="htmlId"
          class="_chip_e7km3_5 _chip_removable_e7km3_62  dummyClassname"
          lang="nb"
          data-testid="123ID"
  >
    Trondheim
    <svg class="_systemIcon_6zbcy_2 _systemIcon_small_6zbcy_5 _icon_e7km3_77"
         viewbox="0 0 24 24"
         focusable="false"
         aria-labelledby="svgtitle-:r1:"
         aria-hidden="false"
         role="img"
    >
      <title id="svgtitle-:r1:">
        Fjern filter TODO
      </title>
      <path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z">
      </path>
    </svg>
  </button>
</div>
`;

exports[`Tester/Chips/Removable WithCloseIconAltText play-test 1`] = `
<div data-test-block="true">
  <button type="button"
          class="_chip_e7km3_5 _chip_removable_e7km3_62"
  >
    Trondheim
    <svg class="_systemIcon_6zbcy_2 _systemIcon_small_6zbcy_5 _icon_e7km3_77"
         viewbox="0 0 24 24"
         focusable="false"
         aria-labelledby="svgtitle-:r4:"
         aria-hidden="false"
         role="img"
    >
      <title id="svgtitle-:r4:">
        Fjern filter TODO
      </title>
      <path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z">
      </path>
    </svg>
  </button>
</div>
`;

exports[`Tester/Chips/Removable WithRef play-test 1`] = `
<div data-test-block="true">
  <button type="button"
          class="_chip_e7km3_5 _chip_removable_e7km3_62"
          id="dummyIdForwardedFromRef"
  >
    Trondheim
    <svg class="_systemIcon_6zbcy_2 _systemIcon_small_6zbcy_5 _icon_e7km3_77"
         viewbox="0 0 24 24"
         focusable="false"
         aria-labelledby="svgtitle-:r0:"
         aria-hidden="false"
         role="img"
    >
      <title id="svgtitle-:r0:">
        Fjern filter TODO
      </title>
      <path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z">
      </path>
    </svg>
  </button>
</div>
`;

exports[`Tester/Chips/Removable WithSizeSmall smoke-test 1`] = `
<div data-test-block="true">
  <button type="button"
          class="_chip_e7km3_5 _chip_removable_e7km3_62 _chip_small_e7km3_51"
  >
    Trondheim
    <svg class="_systemIcon_6zbcy_2 _systemIcon_small_6zbcy_5 _icon_e7km3_77"
         viewbox="0 0 24 24"
         focusable="false"
         aria-labelledby="svgtitle-:r3:"
         aria-hidden="false"
         role="img"
    >
      <title id="svgtitle-:r3:">
        Fjern filter TODO
      </title>
      <path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z">
      </path>
    </svg>
  </button>
</div>
`;
